build/docker-image/self-signer-cert-utility/Dockerfile

```diff
@@ -18,13 +19,16 @@ COPY cmd/ cmd/
 COPY pkg/ pkg/
 
 # Build the binary self-signer utility
-RUN go build -a -installsuffix cgo -o self-signer cmd/main.go
+RUN go build -o self-signer cmd/main.go
 
-# Install the cockroach CLI
-ADD https://binaries.cockroachdb.com/cockroach-v23.1.2.linux-amd64.tgz .
-RUN tar -zxvf cockroach-v23.1.2.linux-amd64.tgz
+# Install the cockroach binary
+RUN if [ "$TARGETPLATFORM" = "linux/amd64" ]; then GOARCH=amd64; elif [ "$TARGETPLATFORM" = "linux/arm64" ]; then GOARCH=arm64; else GOARCH=amd64; fi && \
```

Comment:
Just a minor point but if you were to add `TARGETOS` and `TARGETARCH` args then you can set `GOOS` and `GOARCH` to those values, respectively. Those args are passed in by `docker buildx` (and by `nerdctl build`) intrinsically.

I usually include the following block in my "build" stage so I see what's going on a bit more. I'm not suggesting this change but just putting it here for awareness. You can see where I set `GOOS` and `GOARCH`.

```dockerfile
ARG BUILDOS
ARG BUILDPLATFORM
ARG BUILDARCH
ARG BUILDVARIANT
ARG TARGETPLATFORM
ARG TARGETOS
ARG TARGETARCH
ARG TARGETVARIANT
ARG BUILD_VERSION
ARG BUILD_COMMIT

RUN echo "Container image build platform details :: "
RUN echo "BUILDPLATFORM.....: $BUILDPLATFORM"
RUN echo "BUILDOS...........: $BUILDOS"
RUN echo "BUILDARCH.........: $BUILDARCH"
RUN echo "BUILDVARIANT......: $BUILDVARIANT"
RUN echo "Container image target platform details :: "
RUN echo "TARGETPLATFORM....: $TARGETPLATFORM"
RUN echo "TARGETOS..........: $TARGETOS"
RUN echo "TARGETARCH........: $TARGETARCH"
RUN echo "TARGETVARIANT.....: $TARGETVARIANT"

ENV GOOS=$TARGETOS
ENV GOARCH=$TARGETARCH
```